gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java

```diff
@@ -952,4 +956,62 @@ public void testListDnsRecords() {
       clear();
     }
   }
+
+  @Test
+  public void testListZoneBatch() {
+    DNS.create(ZONE1);
+    DNS.create(ZONE_EMPTY_DESCRIPTION);
+    // the mock is used to check that the callbacks were not skipped
+    final RuntimeException callbackCalled = EasyMock.createStrictMock(RuntimeException.class);
+    EasyMock.expect(callbackCalled.initCause(null)).andReturn(callbackCalled);
+    EasyMock.expect(callbackCalled.initCause(null)).andReturn(callbackCalled);
+    EasyMock.expect(callbackCalled.initCause(null)).andReturn(callbackCalled);
+    EasyMock.replay(callbackCalled);
+    DnsBatch.Callback<Page<Zone>> callback = new DnsBatch.Callback<Page<Zone>>() {
+      @Override
+      public void success(Page<Zone> output, DnsBatch.Request request) {
+        List<Zone> zones = Lists.newArrayList(DNS.listZones().iterateAll());
```

Comment:
What about [Sets.newHashSet](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Sets.html#newHashSet(java.util.Iterator))? Also applies to the next callback.